core/src/main/java/io/questdb/griffin/engine/functions/groupby/MaxDoubleGroupByFunction.java

```diff
@@ -75,6 +75,9 @@ public void setNull(MapValue mapValue) {
 
```

Comment:
Could you please add tests to cover lines changed in Min|Max Int|Long|Double|Timestamp ?

Reply:
https://github.com/questdb/questdb/pull/2388/files#diff-bbb9104cd6bad8237fc1649a7cb3d98359a2d7db78c2ac4259f5db01c70168a7 this is essentially covered here, but if need be I can add more test cases directly to the respective group by functions

Reply:
Sorry, I should've been more clear about this . I ran GroupByTest with coverage and don't see these blocks covered : 
```
if( rec == null) {
   return arg.getDouble(null); //this line is red
}
`

Reply:
Ah - I see. I think this might now actually be unnecessary, as I solved it by forcing it to return isConstant = false. This, in turn, makes sure that (at least in my test cases) they are not called with a "null" argument. 

However, there is still a room for improvement here - the default isConstant for unary operations is to return true if the first arg is a constant. However, when used with groupby, a function returning isConstant currently will lead to a downstream crash as:


In https://github.com/Zahlii/questdb/blob/feat/string-agg/core/src/main/java/io/questdb/griffin/FunctionParser.java#L250 , the function will be converted to a constant (and this would cause the getRecord(null) call above). This part of the bug was essentially fixed with the now untested code. However, further downstream, this assertion here https://github.com/Zahlii/questdb/blob/feat/string-agg/core/src/main/java/io/questdb/griffin/engine/groupby/GroupByUtils.java#L79 is going to fail, as in this case our function is a ConstantFunction (as opposed to a group by function). Ultimately I think this last assertion needs to be reworked so it can actually handle constant functions to avoid unnecessary memory allocations and increase performance, but until then the current code is the hacky way around it, but the null checks could be removed.

Reply:
As an example, select last(0.0) if LastDoubleGroupByFucntion returns isConstant = true, the following would happen: 
![image](https://user-images.githubusercontent.com/218582/186222515-08efaa9f-f6e3-4f84-b1a5-dc6e1bf44f0c.png)

Reply:
Latest changes don't look correct to me . Please consider the following : 

```sql
create table test as (select x from long_sequence(10)); 

select sum(1) from test;
--returns 1  but should be 10

select sum(1) from test where 1=0;
--returns 1 but should be null

select max(1) from test where 1=0; 
--returns 1 but should be null

select first(1) from test where 1=0;
--returns 1 but should be null
```

Aggregate function output depends on input **and** number of rows to process . 
The only case where agg functions could be treated as constant and evaluated before processing any rows is when their arguments are constant and evaluate to null . 

```sql
select sum(null) from test;
select avg(null) from test;
select min(null) from test;
select max(null) from test;
select sum(null) from test where 1=0;
select avg(null) from test where 1=0;
select min(null) from test where 1=0;
select max(null) from test where 1=0;
--all return null
```

count() is a special case and would return 0. 
I doubt it would be useful to optimize such 'narrow' case .

Reply:
I see, will roll that part back. However, for me that indicates that per definition ALL groupbyfucntions should return isConstant false - currently, this isn't correctly maintained, and maybe it would make sense to actually convert groupbyFucntion to an abstract class returning isConstant = false, in order to avoid every subclass to implement this?

Reply:
Making GroupByFunction wouldn't work because all of those functions already inherit from different base classes, e.g. 
```code
public class MaxFloatGroupByFunction extends FloatFunction implements GroupByFunction, UnaryFunction
```

Maybe adding GroupByUnaryFunction wtih overriden methods would work .

Reply:
I'll keep this as is for now, I hope the test cases now cover all data types